src/com/epam/collections/utils/StudentsDynamicArray.java

```diff
@@ -0,0 +1,58 @@
+package com.epam.collections.utils;
+
+import com.epam.collections.interfaces.DynamicArray;
+import com.epam.collections.models.Student;
+
+import java.util.Arrays;
+
+public class StudentsDynamicArray implements DynamicArray {
+    private final int DEFAULT_SIZE = 10;
+    private final int EXPANSION_COEFFICIENT = 2;
+    private int index = 0;
+    private Student[] students;
+
+    public StudentsDynamicArray() {
+        students = new Student[DEFAULT_SIZE];
+    }
+
+    @Override
+    public void add(Student student) {
+        if (index == students.length) {
+            students = Arrays.copyOf(students, students.length * EXPANSION_COEFFICIENT);
+        }
+        students[index++] = student;
+    }
+
+    @Override
+    public boolean remove(Student student) {
+        boolean found = false;
+        for (int j = 0, i = 0; i < size(); ++i) {
+            if (students[i].equals(student) && !found) {
```

Comment:
Looks good!
As a side note, once `found` becomes `true` there is no need to evaluate `students[i].equals(student)` condition, so it would be better to replace the order of conditions inside the `if` statement